docs/source/run/parameters.rst

```diff
@@ -749,3 +718,53 @@ Field diagnostics
 
 * ``<diag name> or diagnostic.patch_hi`` (3 `float`) optional (default `infinity infinity infinity`)
     Upper limit for the diagnostic grid.
+
+In-situ diagnostics
+^^^^^^^^^^^^^^^^^^^
+
+Besides the standard diagnostics, fast in-situ diagnostics are available. They are most useful when beams with a large number of particles is used, because then the important moments can be calculated by GPU and significantly reduce the post-analysis of the simulation.
+For particle beams, they can be used to calculate the main characterizing beam parameters (width, energy spread, emittance, etc.). Additionally, the plasma particle properties (e.g, the temperature) can be calculated.
+
+The in-situ quantities are calculated per slice and together with the weights per slice, all important quantities can be calculated.
+
+For particle beams, the following quantities are calculated per slice and stored:
+``sum(w), [x], [x^2], [y], [y^2], [z], [z^2], [ux], [ux^2], [uy], [uy^2], [uz], [uz^2], [x*ux], [y*uy], [z*uz], [ga], [ga^2], np``.
+For plasma particles, the following quantities are calculated per slice and stored:
+``sum(w), [x], [x^2], [y], [y^2], [ux], [ux^2], [uy], [uy^2], [uz], [uz^2], [ga], [ga^2], np``.
```

Comment:
We could also use exactly the same parameters for the plasma, although I do not see any use case for `[x*ux], [y*uy], [z*uz],` for a plasma at the moment.

Reply:
No strong opinion, either works for me.

Reply:
Then let's keep it as is for now